src/app/auth/login.ts

```diff
@@ -113,6 +132,331 @@ export const loginWithPhoneCookie = async ({
+  ///               1. login with phone account and create an invoice (save this invoice)
+  ///               2. logout of the phone account
+  ///               3. login with device account (it should auto login)
+  ///               4  Pay the invoice with max wallet amount.
+  ///               5. Logout of the device account
+  ///               6. Log into the phone account and check for the funds
+  ///
+  ///         * FUTURE USE CASE: is it a risk to have txn history persist if user sells phone?
+  ///         * can a user manually sweep funds to new account then click a btn in the mobile app to delete device account?
+  ///               /// if device account bal is 0 and user clicks nuke account then delete kratos and mongo entries with
+  ///                      mutation nukeDeviceAccountMutation ?
+  ///
+  /// Scenario 2) Happy Path - User logs in with device jwt, no phone account exists, upgrade device to phone account
+  ///             a. update kratos => update schema to phone_no_password_v0, remove device trait
+  ///             b. mongo (user) => update user collection and remove device field, add phone
+  ///             c. mongo (account) => update account to level 1
+  ///
+  /// Scenario 3) Unhappy path - User sells phone and forgets to sweep funds to phone account
+  ///             Option 1 - too bad, we can't help them
+  ///             Option 2 - create some kind of recovery code process?
+
+  /// Scenario 1 - does phone account already exist?
+  const phoneAccount = await UsersRepository().findByPhone(phone)
+  let hasPhoneAccount = true
+  if (phoneAccount instanceof CouldNotFindUserFromPhoneError) {
+    hasPhoneAccount = false
+  } else {
+    if (phoneAccount instanceof Error) return phoneAccount
+  }
+  if (hasPhoneAccount) {
+    // is there still txns left over on the device account?
+    const deviceWallets = await WalletsRepository().listByAccountId(account.id)
+    if (deviceWallets instanceof Error) return deviceWallets
+    const ledger = LedgerService()
+    let deviceAccountHasBalance = false
+    for (const wallet of deviceWallets) {
+      const balance = await ledger.getWalletBalance(wallet.id)
+      if (balance instanceof Error) return balance
+      if (balance > 0) {
+        deviceAccountHasBalance = true
+      }
+    }
+    if (deviceAccountHasBalance) {
+      /// Scenario 1.5 - has txns on device account but phone account exists
+      return new PhoneAccountAlreadyExistsNeedToSweepFunds(
+        `Error phone account already exists. You need to manually sweep funds
+        to your phone account. Here are the steps to do this:
+        1. login with phone account and create an invoice (save this invoice)
+        2. logout of the phone account
+        3. login with device account (click "Explore Wallet Instead" button)
+        4  Pay the invoice with max wallet amount.
+        5. Logout of the device account
+        6. Log into the phone account and check for the funds`,
```

Comment:
this should be in the frontend eventually

Reply:
added it as an error type here https://github.com/GaloyMoney/galoy/pull/2071/commits/9a6050ee9bcc834111ff442c34cec9ca932bee25 for @UncleSamtoshi to handle on the frontend